docs/changes/io.fits/16581.bugfix.rst

```diff
@@ -0,0 +1,3 @@
+Skip an un-portable hack for Windows if NumPy 2.0 or newer is installed.
+This fixes a crash that occurred for files oopened via
+``fits.open(..., mode='update')``.
```

Comment:
Change log entries should describe to users what has changed in enough detail that looking up the actual code changes or any discussions on GitHub should not be required. If I try to read the two sentences here without any other context I get the question if skipping a hack fixes a crash then does that mean that the hack caused the crashes? If so then why was the hack there in the first place?

Reply:
> does that mean that the hack caused the crashes ?

This would be the correct conclusion.

> If so then why was the hack there in the first place?

It's already hard for us to know that. What would you suggest we say instead ?

Reply:
I'm not familiar with `io.fits` and I'm not familiar with Windows either, so I don't really know what to say. I'm just pointing out that the change log is meant to be for the benefit of the users so the entries should not confuse them.

Reply:
How about this ?
```
Fixed a crash that occurred for files opened via ``fits.open(..., mode='update')``,
on Windows, and with numpy 2.0 installed.
It is possible, though unlikely, that this patch reveals memory issues
for downstream code run under these exact conditions. Please report any 
regression found.
```

Reply:
Darn, sorry, I should have checked more carefully. That substitution is definitely better. Sigh.